src/main/java/com/swirlds/recordserver/TransactionsService.java

```diff
@@ -0,0 +1,205 @@
+package com.swirlds.recordserver;
+
+import com.swirlds.recordserver.util.QueryParamUtil;
+import io.helidon.config.Config;
+import io.helidon.metrics.api.RegistryFactory;
+import io.helidon.webserver.Routing;
+import io.helidon.webserver.ServerRequest;
+import io.helidon.webserver.ServerResponse;
+import io.helidon.webserver.Service;
+import jakarta.json.*;
+import org.apache.pinot.client.Connection;
+import org.apache.pinot.client.ConnectionFactory;
+import org.apache.pinot.client.PreparedStatement;
+import org.apache.pinot.client.Request;
+import org.apache.pinot.client.ResultSet;
+import org.apache.pinot.client.ResultSetGroup;
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import static com.swirlds.recordserver.util.QueryParamUtil.parseLimitQueryString;
+import static com.swirlds.recordserver.util.Utils.parseFromColumn;
+import static com.swirlds.recordserver.util.Utils.parseArrayFromColumn;
+
+/**
+ * A service for transactions API
+ */
+
+public class TransactionsService implements Service {
+
+    public enum transactionType {CREDIT,DEBIT };
+    private static final Logger LOGGER = Logger.getLogger(TransactionsService.class.getName());
+    private static final JsonBuilderFactory JSON = Json.createBuilderFactory(Collections.emptyMap());
+    private final MetricRegistry registry = RegistryFactory.getInstance().getRegistry(MetricRegistry.Type.APPLICATION);
+    private final Counter accessCtr = registry.counter("accessctr");
+    private final Connection pinotConnection;
+
+    public TransactionsService(Config config) {
+        this.pinotConnection = ConnectionFactory.fromHostList(config.get("pinot-broker").asString().orElse("pinot-broker:8099"));
+    }
+
+    /**
+     * A service registers itself by updating the routing rules.
+     *
+     * @param rules the routing rules.
+     */
+
+    @Override
+    public void update(Routing.Rules rules) {
+        rules.get("/", this::getTransactionIdMessageHandler);
+    }
+
+    private void getTransactionIdMessageHandler(ServerRequest request, ServerResponse response) {
+        System.out.println("The request path is " + request.queryParams());
+        if(!request.queryParams().toMap().containsKey("transactionId")) {
+             getTransactionsMessageHandler(request, response);
+             return;
+        }
+        final Optional<String> nonceParam = request.queryParams().first("nonce");
+        final Optional<String> scheduledParam = request.queryParams().first("scheduled");
+        final Optional<String> transactionIdParam = request.queryParams().first("transactionId");
+
+        // build and execute query
+        final List<QueryParamUtil.WhereClause> whereClauses = new ArrayList<>();
+
+        for (var where: whereClauses) {
```

Comment:
shouldn't this loop (lines 71-73) come after lines 75-77?  Where it is now, you are just iterating over an empty list.

Reply:
Left it in there by mistake.